staff_features/resources/step_definitions/resource_archival_object_create.rb

```diff
@@ -0,0 +1,194 @@
+
+  url_parts = current_url.split('/')
+  url_parts.pop
+  @resource_id = url_parts.pop
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 1 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  fill_in 'Component Unique Identifier', with: "Archival Object 1 #{@uuid}"
+  check 'Publish?'
+  check 'Restrictions Apply?'
+  fill_in 'Repository Processing Note', with: "Repository Processing Note #{@uuid}"
+
+  click_on 'Add Language'
+  fill_in 'Language', with: 'English'
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
+  fill_in 'Script', with: 'adlam'
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
+
+  click_on 'Add Date'
+  select 'Single', from: 'archival_object_dates__0__date_type_'
+  fill_in 'archival_object_dates__0__begin_', with: '2000-01-01'
+
+  click_on 'Add Extent'
+  fill_in 'Number', with: @uuid
+  select 'Cassettes', from: 'archival_object_extents__0__extent_type_'
+
+  click_on 'Add Agent Link'
+  select 'Creator', from: 'archival_object_linked_agents__0__role_'
+  fill_in 'archival_object_linked_agents__0__title_', with: "Accession #{@uuid} Agent Title"
+  fill_in 'archival_object_linked_agents__0__relator_', with: 'annotator'
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
+  fill_in 'token-input-archival_object_linked_agents__0__ref_', with: 'test_agent'
+  dropdown_items = all('li.token-input-dropdown-item2')
+  dropdown_items.first.click
+
+  click_on 'Add Accession Link'
+  fill_in 'token-input-accession_links_', with: 'test_accession'
+  dropdown_items = all('li.token-input-dropdown-item2')
+  dropdown_items.first.click
+
+  click_on 'Add Subject'
+  fill_in 'token-input-archival_object_subjects__0__ref_', with: 'test_subject'
+  dropdown_items = all('li.token-input-dropdown-item2')
+  dropdown_items.first.click
+
+  click_on 'Add Note'
+  note_type = find('#notes select')
+  note_type.select 'Abstract'
+  fill_in 'archival_object_notes__0__persistent_id_', with: "Persistent ID #{@uuid}"
+  fill_in 'archival_object_notes__0__label_', with: "Label #{@uuid}"
+  check 'archival_object_notes__0__publish_'
+  page.execute_script("$('#archival_object_notes__0__content__0_').data('CodeMirror').setValue('Content #{@uuid}')")
+
+  click_on 'Add External Document'
+  fill_in 'archival_object_external_documents__0__title_', with: "External Document Title #{@uuid}"
+  fill_in 'archival_object_external_documents__0__location_', with: "External Document Location #{@uuid}"
+  check 'archival_object_external_documents__0__publish_'
+
+  click_on 'Add Rights Statement'
+  select 'Copyright', from: 'archival_object_rights_statements__0__rights_type_'
+  fill_in 'archival_object_rights_statements__0__jurisdiction_', with: 'andorra'
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
```

Comment:
do the .typeahead-typeahead-long.dropdown-menu and li.token-input-dropdown-item2 elements get rewritten with clicks on 'Add Accession Link', 'Add Subject' etc? The test repeatedly selects the first item from what looks like these same elements over and over again after clicking on links and filling in data.

Reply:
> The test repeatedly selects the first item from what looks like these same elements over and over again

@balmas They're not the same elements. They're related to very common "linker" dropdowns, which link one record to another and are found in many record subforms. The above code looks as I expect.

Reply:
I believe that what you mean here is to extract common functionality to a helper function, is that correct?
It could be, as from what I have seen so far there are various places where search and selection could occur, however, there are quite some variations on the form field ids that differ, even if the functionality is the same from the UI perspective.